tox.ini

```diff
@@ -1,26 +1,46 @@
 [tox]
 requires =
     tox>=4
-env_list = coverage, build, firmware, docs, format, type, py{39,310,311,312}
+env_list = coverage, build, firmware, docs, format, type, py{39,310,311,312}, virtual
 
 # Use this deps to install custom requirements in develop.
 # For instance, to install ingenialink from a custom branch.
 # It should be empty in production.
 [develop]
-    deps = git+https://github.com/ingeniamc/ingenialink-python@develop
+    ingenialink = git+https://github.com/ingeniamc/ingenialink-python@develop
 
 [testenv]
 description = run unit tests
-deps = 
-    {[develop]deps}
+deps =
+    {[develop]ingenialink}
+    pytest==7.0.1
+    pytest-cov==2.12.1
+    pytest-mock==3.6.1
+    ping3==4.0.3
+    pytest-html==3.1.1
+    pytest-console-scripts==1.4.1
+    matplotlib==3.8.2
+commands =
+    python -m pytest {posargs:tests}
+
+[testenv:virtual]
+allowlist_externals = *
+passenv = *
+description = run unit tests
+deps =
+    {[develop]ingenialink}
     pytest==7.0.1
     pytest-cov==2.12.1
     pytest-mock==3.6.1
     ping3==4.0.3
     pytest-html==3.1.1
     pytest-console-scripts==1.4.1
     matplotlib==3.8.2
+    cython==3.0.7
 commands =
+    git clone git+ssh://git@bitbucket.org/novantamotion/fsoe_master_python
+    git -C fsoe_master_python// checkout 3fa5d29
```

Comment:
I'd make the commit reference parametrizable. You can add it in the [develop] section:

```
[develop]
    ingenialink = git+https://github.com/ingeniamc/ingenialink-python@develop
    fsoe_master_ref = 3fa5d29
```
And then use it like this:
```
git -C fsoe_master_python// checkout {[develop]fsoe_master_ref}
```